Foundation/include/Poco/ProcessRunner.h

```diff
@@ -0,0 +1,189 @@
+//
+// ProcessRunner.h
+//
+// Library: Foundation
+// Package: Processes
+// Module:  ProcessRunner
+//
+// Definition of the ProcessRunner class.
+//
+// Copyright (c) 2023, Applied Informatics Software Engineering GmbH.
+// Aleph ONE Software Engineering d.o.o.,
+// and Contributors.
+//
+// SPDX-License-Identifier:    BSL-1.0
+//
+
+
+#ifndef Foundation_ProcessRunner_INCLUDED
+#define Foundation_ProcessRunner_INCLUDED
+
+
+#include "Poco/Foundation.h"
+#include "Poco/Process.h"
+#include "Poco/ProcessOptions.h"
+#include "Poco/Runnable.h"
+#include "Poco/Thread.h"
+#include "Poco/Format.h"
+#include "Poco/Stopwatch.h"
+#include <atomic>
+#include <vector>
+
+
+namespace Poco {
+
+
+class Foundation_API ProcessRunner: public Poco::Runnable
+	/// ProcessRunner is a wrapper class for `Poco::ProcessHandle.
+	/// It starts and terminates a process with enabled or disabled (default)
+	/// `stdio` pipes, and optionally waits for process PID to be created before
+	/// returning control to the caller. The process is spawned from an
+	/// internal thread. Starting/stopping the process may block up to
+	/// a certain (configurable) period of time.
+	///
+	/// ProcessRunner can hold and control only one process at a time, which
+	/// can be started/stopped multiple times during the ProcessRunner lifetime.
+{
+public:
+	using Args = Poco::Process::Args;
+	using PID = Poco::ProcessHandle::PID;
+
+	static const int NO_OUT = Poco::PROCESS_CLOSE_STDOUT|Poco::PROCESS_CLOSE_STDERR;
+		/// Constant to prevent std out and err from being received from the process.
+
+	ProcessRunner(const std::string& cmd,
+		const Args& args,
+		const std::string& pidFile = "",
+		int options = NO_OUT,
+		int timeout = 10, /*seconds*/
+		bool startProcess = true,
+		const Args& pidArgFmt = {"-p", "--pidfile="});
```

Comment:
the comment that was here still applies, windows has `/` convention for parameters; when I said "disregard windows" I meant for terminating the process, see here:
https://github.com/pocoproject/poco/blob/83d50284e07df356dcb387165d23bcc2b544682d/Util/src/HelpFormatter.cpp#L276-L293

Reply:
Fixed in https://github.com/pocoproject/poco/pull/4225/commits/ce8ba27335a3d558c667a6c89c9f242019aaf22b